### ISSUE TITLE
Source selection drop box does not retrieve the product user configured sources

### ISSUE THREAD

**Issue by mbolo01** [giachello/beoplay#2]
Source selection drop box does not retrieve the product's user configured sources, but all possible sources, even the unused ones or the one that you cannot action from the product itself such as Airplay or Chromecast.
I worked with TLK (https://github.com/tlk/beoplay-macos-remote-gui) to fix this behavior as TLK was initially inspired by legacy beoplay code to write his own (clever OSX Beoplay control tool) so the code was also retrieving all these useless sources.
The sources drop box I'm referring to:
<img width="906" alt="Capture d’écran 2021-01-21 à 15 02 06" src="https://user-images.githubusercontent.com/57586871/105361790-5de25e00-5bfa-11eb-89ff-821c09e85f0e.png">

**Comment by giachello** [comment]
Hi, yes I know about this issue and I fixed it in a new version. I'm waiting for @marton to commit the underlying ha-beoplay module to release the new version

**Comment by mbolo01** [comment]
Great.
Regarding metadata management (e.g cover image, track and album title) for all capable sources, is it under your radar too? For now I'm editing the media_player.py to make sure I get them displayed when using AirPlay or Chromecast sources.

**Comment by giachello** [comment]
That should already be handled. Not sure why it's not. Let me check in the weekend.

**Comment by giachello** [comment]
Ok, try now, I fixed it the sources issues. I haven't looked into the metadata issue. I cannot debug that because I only have a TV which I'm not sure shares metadata.

**Comment by mbolo01** [comment]
I removed the previous code and integrations, that I replaced by the new code and added back integrations: The sources are still the same for me. 
I guess you populate them by looking at "http://<device IP>:8080/BeoZone/Zone/ActiveSources/BeoZone/Zone/Sources" which is not really useful, even B&O gave up in their app looking at them from this angle for their new Halo remote. 
As an example, for my M3, it returns: 
TuneIn, Qplay, Bluetooth, Line-In, Airplay, Deezer, Chromecast built-in, Music stream and Alarm .... Amongst these 9 sources, there are only 5 are sources that you can trigger from the device, so the media-player card, other sources are triggered externally to the product  (Airplay, CC, BT) or by internal setup (Alarm).

Preferred sources are set in the product local list that you configure with Content/Sources/In Local list, these are the sources you really care about for your product, in my case, the M3 only has TuneIn (see attachement).
![IMG_3AF355AA2B23-1](https://user-images.githubusercontent.com/57586871/105879068-0715c380-6002-11eb-961b-8bd8a9e8471a.jpeg)

These sources in local list can be found here:

GET /api/getData?path=settings:/beo/sources/controlledSources&roles=value HTTP/1.1
Host: beosound-1-30478942.local.:80
Content-Type: application/json; charset=utf-8
Accept: */*
User-Agent: BeoplayRemoteGUI/1 CFNetwork/978.4 Darwin/18.7.0 (x86_64)
Accept-Language: fr-fr
Accept-Encoding: gzip, deflate
Connection: keep-alive

Which produces the following output:
[{"controlledSources":{"controlledSources":[{"deviceId":"","**enabled":true,**"sourceId":"radio","enabledExternal":true},{"deviceId":"","enabled":false,"sourceId":"music","enabledExternal":true},{"deviceId":"","enabled":false,"sourceId":"qplay","enabledExternal":true},{"deviceId":"","enabled":false,"sourceId":"bluetooth","enabledExternal":true},{"deviceId":"","enabled":false,"sourceId":"linein","enabledExternal":true},{"deviceId":"","enabled":false,"sourceId":"airplay","enabledExternal":false},{"deviceId":"","enabled":false,"sourceId":"deezer","enabledExternal":false},{"deviceId":"","enabled":false,"sourceId":"googlecast","enabledExternal":false},{"deviceId":"","enabled":false,"sourceId":"dlna","enabledExternal":false},{"deviceId":"","enabled":false,"sourceId":"alarm","enabledExternal":false}]},"type":"controlledSources"}]

Regarding the metadata, I would remove each occurence of :

**if self._source == "AirPlay":
            return None**

in each relevant section of media_player.py  such as:

**@property
    def media_image_url(self):
        """Image url of current playing media."""**

As Airplay effectively sends the metadata that can be exposed by the media_player entity.

On my side I have added more condition to process specific sources of specific products, no real interest for other users, for now, but we can discuss that later.

**Comment by giachello** [comment]
Hi there, thank you for the hints. I don't own a speaker that provides /api/getData so cannot debug that code.
On my end, I did remove sources that have "inUse = false" and that keeps only the used sources of my TV and Beolink device.
For the media image url, I didn't test the function so hard to tell. If you have a modification that works for you, please make the pull request and we can discuss including it in the code.

Thank you for your help in making this plugin better!

**Comment by mbolo01** [comment]
Hi, there. 
Regarding sources, I understand it can be complicated without a capable device. If you have time, have a look at this small (and efficient) MACOS tool which leverages the getData method to pull the speakers user selected sources: https://github.com/tlk/beoplay-macos-remote-gui
Regarding the metadata: I made the pull request, as it is the first time I submit one, hopefully I did it correctly.
Cheers.
